sdl_android_lib/src/com/smartdevicelink/transport/SdlPsm.java

```diff
@@ -170,7 +177,11 @@ private int transitionOnInput(byte rawByte, int state){
 				if(dataLength == 0){
 					return FINISHED_STATE; //We are done if we don't have any payload
 				}
-				payload = new byte[dataLength];
+				try {
```

Comment:
Since there is a check for a v1 packet right before this, we know only a v1 packet could enter this if statement and therefore would have to be less than or equal to V1_V2_MTU_SIZE - V1_HEADER_SIZE